[Bug]: portable story tests are broken due to next/dist imports without .js extension

### Describe the bug

```
Error: Cannot find module '<REDUCTED>/node_modules/.pnpm/@storybook+nextjs@10.1.9_esbuild@0.27.2_next@16.0.10_@babel+core@7.28.5_react-dom@19.2._fb92b71e4bbbffb1b122e48ebb3990e6/node_modules/next/dist/client/components/navigation' imported from <REDUCTED>/node_modules/.pnpm/@storybook+nextjs@10.1.9_esbuild@0.27.2_next@16.0.10_@babel+core@7.28.5_react-dom@19.2._fb92b71e4bbbffb1b122e48ebb3990e6/node_modules/@storybook/nextjs/dist/export-mocks/navigation/index.js
Did you mean to import "next/dist/client/components/navigation.js"?
```

### Reproduction link

https://github.com/lab-yue/reproduction-portable-stories-next

### Reproduction steps

Follow README in the repo.

### System

```bash
│  System:
│  OS: macOS 26.2
│  CPU: (16) arm64 Apple M4 Max
│  Shell: 5.9 - /bin/zsh
│  Binaries:
│  Node: 24.12.0 - /Users/yue/Library/pnpm/node
│  Yarn: 1.22.22 - /Users/yue/Library/pnpm/yarn
│  npm: 11.6.2 - /Users/yue/Library/pnpm/npm
│  pnpm: 10.25.0 - /Users/yue/Library/pnpm/pnpm <----- active
│  Browsers:
│  Chrome: 143.0.7499.146
│  Safari: 26.2
│  npmPackages:
│  @storybook/addon-a11y: ^10.1.9 => 10.1.9
│  @storybook/addon-docs: ^10.1.9 => 10.1.9
│  @storybook/addon-onboarding: ^10.1.9 => 10.1.9
│  @storybook/addon-vitest: ^10.1.9 => 10.1.9
│  @storybook/nextjs: ^10.1.9 => 10.1.9
│  eslint-plugin-storybook: ^10.1.9 => 10.1.9
│  storybook: ^10.1.9 => 10.1.9
```

### Additional context

_No response_

--- a/code/frameworks/nextjs-vite/src/export-mocks/headers/cookies.ts
+++ b/code/frameworks/nextjs-vite/src/export-mocks/headers/cookies.ts
@@ -5,7 +5,7 @@
 // @ts-ignore we must ignore types here as during compilation they are not generated yet
 import { headers } from '@storybook/nextjs-vite/headers.mock';
 
-import { RequestCookies } from 'next/dist/compiled/@edge-runtime/cookies';
+import { RequestCookies } from 'next/dist/compiled/@edge-runtime/cookies/index.js';
 import { fn } from 'storybook/test';
 
 class RequestCookiesMock extends RequestCookies {

--- a/code/frameworks/nextjs-vite/src/export-mocks/headers/headers.ts
+++ b/code/frameworks/nextjs-vite/src/export-mocks/headers/headers.ts
@@ -1,4 +1,4 @@
-import { HeadersAdapter } from 'next/dist/server/web/spec-extension/adapters/headers';
+import { HeadersAdapter } from 'next/dist/server/web/spec-extension/adapters/headers.js';
 import { fn } from 'storybook/test';
 
 class HeadersAdapterMock extends HeadersAdapter {

--- a/code/frameworks/nextjs-vite/src/export-mocks/headers/index.ts
+++ b/code/frameworks/nextjs-vite/src/export-mocks/headers/index.ts
@@ -1,9 +1,9 @@
-import { draftMode as originalDraftMode } from 'next/dist/server/request/draft-mode';
-import * as headers from 'next/dist/server/request/headers';
+import { draftMode as originalDraftMode } from 'next/dist/server/request/draft-mode.js';
+import * as headers from 'next/dist/server/request/headers.js';
 import { fn } from 'storybook/test';
 
 // re-exports of the actual module
-export * from 'next/dist/server/request/headers';
+export * from 'next/dist/server/request/headers.js';
 
 // mock utilities/overrides (as of Next v14.2.0)
 export { headers } from './headers';

--- a/code/frameworks/nextjs-vite/src/export-mocks/navigation/index.ts
+++ b/code/frameworks/nextjs-vite/src/export-mocks/navigation/index.ts
@@ -1,8 +1,8 @@
 import { NextjsRouterMocksNotAvailable } from 'storybook/internal/preview-errors';
 
-import * as actual from 'next/dist/client/components/navigation';
-import { getRedirectError } from 'next/dist/client/components/redirect';
-import { RedirectStatusCode } from 'next/dist/client/components/redirect-status-code';
+import * as actual from 'next/dist/client/components/navigation.js';
+import { RedirectStatusCode } from 'next/dist/client/components/redirect-status-code.js';
+import { getRedirectError } from 'next/dist/client/components/redirect.js';
 import type { Mock } from 'storybook/test';
 import { fn } from 'storybook/test';
 
@@ -57,7 +57,7 @@ export const getRouter = () => {
 };
 
 // re-exports of the actual module
-export * from 'next/dist/client/components/navigation';
+export * from 'next/dist/client/components/navigation.js';
 
 // mock utilities/overrides (as of Next v14.2.0)
 export const redirect = fn(

--- a/code/frameworks/nextjs-vite/src/export-mocks/router/index.ts
+++ b/code/frameworks/nextjs-vite/src/export-mocks/router/index.ts
@@ -1,6 +1,6 @@
 import { NextjsRouterMocksNotAvailable } from 'storybook/internal/preview-errors';
 
-import singletonRouter, * as originalRouter from 'next/dist/client/router';
+import singletonRouter, * as originalRouter from 'next/dist/client/router.js';
 import type { NextRouter, SingletonRouter } from 'next/router';
 import type { Mock } from 'storybook/test';
 import { fn } from 'storybook/test';
@@ -108,7 +108,7 @@ export const getRouter = () => {
 };
 
 // re-exports of the actual module
-export * from 'next/dist/client/router';
+export * from 'next/dist/client/router.js';
 export default singletonRouter;
 
 // mock utilities/overrides (as of Next v14.2.0)

--- a/code/frameworks/nextjs-vite/src/head-manager/head-manager-provider.tsx
+++ b/code/frameworks/nextjs-vite/src/head-manager/head-manager-provider.tsx
@@ -1,8 +1,8 @@
 import type { PropsWithChildren } from 'react';
 import React, { useMemo } from 'react';
 
-import initHeadManager from 'next/dist/client/head-manager';
-import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context.shared-runtime';
+import initHeadManager from 'next/dist/client/head-manager.js';
+import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context.shared-runtime.js';
 
 type HeadManagerValue = {
   updateHead?: ((state: JSX.Element[]) => void) | undefined;

--- a/code/frameworks/nextjs-vite/src/preview.tsx
+++ b/code/frameworks/nextjs-vite/src/preview.tsx
@@ -11,7 +11,7 @@ import { createNavigation } from '@storybook/nextjs-vite/navigation.mock';
 // @ts-ignore we must ignore types here as during compilation they are not generated yet
 import { createRouter } from '@storybook/nextjs-vite/router.mock';
 
-import { isNextRouterError } from 'next/dist/client/components/is-next-router-error';
+import { isNextRouterError } from 'next/dist/client/components/is-next-router-error.js';
 
 import { HeadManagerDecorator } from './head-manager/decorator';
 import { ImageDecorator } from './images/decorator';

--- a/code/frameworks/nextjs-vite/src/routing/app-router-provider.tsx
+++ b/code/frameworks/nextjs-vite/src/routing/app-router-provider.tsx
@@ -12,13 +12,13 @@ import {
   AppRouterContext,
   GlobalLayoutRouterContext,
   LayoutRouterContext,
-} from 'next/dist/shared/lib/app-router-context.shared-runtime';
+} from 'next/dist/shared/lib/app-router-context.shared-runtime.js';
 import {
   PathParamsContext,
   PathnameContext,
   SearchParamsContext,
-} from 'next/dist/shared/lib/hooks-client-context.shared-runtime';
-import { PAGE_SEGMENT_KEY } from 'next/dist/shared/lib/segment';
+} from 'next/dist/shared/lib/hooks-client-context.shared-runtime.js';
+import { PAGE_SEGMENT_KEY } from 'next/dist/shared/lib/segment.js';
 
 import type { RouteParams } from './types';
 

--- a/code/frameworks/nextjs-vite/src/routing/decorator.tsx
+++ b/code/frameworks/nextjs-vite/src/routing/decorator.tsx
@@ -2,7 +2,7 @@ import * as React from 'react';
 
 import type { Addon_StoryContext } from 'storybook/internal/types';
 
-import { RedirectBoundary } from 'next/dist/client/components/redirect-boundary';
+import { RedirectBoundary } from 'next/dist/client/components/redirect-boundary.js';
 
 import { AppRouterProvider } from './app-router-provider';
 import { PageRouterProvider } from './page-router-provider';

--- a/code/frameworks/nextjs-vite/src/routing/page-router-provider.tsx
+++ b/code/frameworks/nextjs-vite/src/routing/page-router-provider.tsx
@@ -8,7 +8,7 @@ import React from 'react';
 // @ts-ignore we must ignore types here as during compilation they are not generated yet
 import { getRouter } from '@storybook/nextjs-vite/router.mock';
 
-import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';
+import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime.js';
 
 export const PageRouterProvider: React.FC<PropsWithChildren> = ({ children }) => (
   <RouterContext.Provider value={getRouter()}>{children}</RouterContext.Provider>

--- a/code/frameworks/nextjs/src/compatibility/draft-mode.compat.ts
+++ b/code/frameworks/nextjs/src/compatibility/draft-mode.compat.ts
@@ -1,2 +1,2 @@
 // @ts-expect-error Compatibility for Next 14
-export { draftMode } from 'next/dist/client/components/headers';
+export { draftMode } from 'next/dist/client/components/headers.js';

--- a/code/frameworks/nextjs/src/export-mocks/cache/index.ts
+++ b/code/frameworks/nextjs/src/export-mocks/cache/index.ts
@@ -1,5 +1,5 @@
-import { unstable_cache } from 'next/dist/server/web/spec-extension/unstable-cache';
-import { unstable_noStore } from 'next/dist/server/web/spec-extension/unstable-no-store';
+import { unstable_cache } from 'next/dist/server/web/spec-extension/unstable-cache.js';
+import { unstable_noStore } from 'next/dist/server/web/spec-extension/unstable-no-store.js';
 import { fn } from 'storybook/test';
 
 // mock utilities/overrides (as of Next v14.2.0)

--- a/code/frameworks/nextjs/src/export-mocks/headers/cookies.ts
+++ b/code/frameworks/nextjs/src/export-mocks/headers/cookies.ts
@@ -5,7 +5,7 @@
 // @ts-ignore we must ignore types here as during compilation they are not generated yet
 import { headers } from '@storybook/nextjs/headers.mock';
 
-import { RequestCookies } from 'next/dist/compiled/@edge-runtime/cookies';
+import { RequestCookies } from 'next/dist/compiled/@edge-runtime/cookies/index.js';
 import { fn } from 'storybook/test';
 
 class RequestCookiesMock extends RequestCookies {

--- a/code/frameworks/nextjs/src/export-mocks/headers/headers.ts
+++ b/code/frameworks/nextjs/src/export-mocks/headers/headers.ts
@@ -1,4 +1,4 @@
-import { HeadersAdapter } from 'next/dist/server/web/spec-extension/adapters/headers';
+import { HeadersAdapter } from 'next/dist/server/web/spec-extension/adapters/headers.js';
 import { fn } from 'storybook/test';
 
 class HeadersAdapterMock extends HeadersAdapter {

--- a/code/frameworks/nextjs/src/export-mocks/headers/index.ts
+++ b/code/frameworks/nextjs/src/export-mocks/headers/index.ts
@@ -1,8 +1,8 @@
-import * as headers from 'next/dist/server/request/headers';
+import * as headers from 'next/dist/server/request/headers.js';
 import { fn } from 'storybook/test';
 
 // re-exports of the actual module
-export * from 'next/dist/server/request/headers';
+export * from 'next/dist/server/request/headers.js';
 
 // mock utilities/overrides (as of Next v14.2.0)
 export { headers } from './headers';

--- a/code/frameworks/nextjs/src/export-mocks/navigation/index.ts
+++ b/code/frameworks/nextjs/src/export-mocks/navigation/index.ts
@@ -1,8 +1,8 @@
 import { NextjsRouterMocksNotAvailable } from 'storybook/internal/preview-errors';
 
-import * as actual from 'next/dist/client/components/navigation';
-import { getRedirectError } from 'next/dist/client/components/redirect';
-import { RedirectStatusCode } from 'next/dist/client/components/redirect-status-code';
+import * as actual from 'next/dist/client/components/navigation.js';
+import { RedirectStatusCode } from 'next/dist/client/components/redirect-status-code.js';
+import { getRedirectError } from 'next/dist/client/components/redirect.js';
 import type { Mock } from 'storybook/test';
 import { fn } from 'storybook/test';
 
@@ -57,7 +57,7 @@ export const getRouter = () => {
 };
 
 // re-exports of the actual module
-export * from 'next/dist/client/components/navigation';
+export * from 'next/dist/client/components/navigation.js';
 
 // mock utilities/overrides (as of Next v14.2.0)
 export const redirect = fn(

--- a/code/frameworks/nextjs/src/export-mocks/router/index.ts
+++ b/code/frameworks/nextjs/src/export-mocks/router/index.ts
@@ -1,6 +1,6 @@
 import { NextjsRouterMocksNotAvailable } from 'storybook/internal/preview-errors';
 
-import singletonRouter, * as originalRouter from 'next/dist/client/router';
+import singletonRouter, * as originalRouter from 'next/dist/client/router.js';
 import type { NextRouter, SingletonRouter } from 'next/router';
 import type { Mock } from 'storybook/test';
 import { fn } from 'storybook/test';
@@ -108,7 +108,7 @@ export const getRouter = () => {
 };
 
 // re-exports of the actual module
-export * from 'next/dist/client/router';
+export * from 'next/dist/client/router.js';
 export default singletonRouter;
 
 // mock utilities/overrides (as of Next v14.2.0)

--- a/code/frameworks/nextjs/src/head-manager/head-manager-provider.tsx
+++ b/code/frameworks/nextjs/src/head-manager/head-manager-provider.tsx
@@ -1,8 +1,8 @@
 import type { PropsWithChildren } from 'react';
 import React, { useMemo } from 'react';
 
-import initHeadManager from 'next/dist/client/head-manager';
-import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context.shared-runtime';
+import initHeadManager from 'next/dist/client/head-manager.js';
+import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context.shared-runtime.js';
 
 type HeadManagerValue = {
   updateHead?: ((state: JSX.Element[]) => void) | undefined;

--- a/code/frameworks/nextjs/src/preview.tsx
+++ b/code/frameworks/nextjs/src/preview.tsx
@@ -12,7 +12,7 @@ import { createNavigation } from '@storybook/nextjs/navigation.mock';
 // @ts-ignore we must ignore types here as during compilation they are not generated yet
 import { createRouter } from '@storybook/nextjs/router.mock';
 
-import { isNextRouterError } from 'next/dist/client/components/is-next-router-error';
+import { isNextRouterError } from 'next/dist/client/components/is-next-router-error.js';
 
 import { HeadManagerDecorator } from './head-manager/decorator';
 import { ImageDecorator } from './images/decorator';

--- a/code/frameworks/nextjs/src/routing/app-router-provider.tsx
+++ b/code/frameworks/nextjs/src/routing/app-router-provider.tsx
@@ -12,13 +12,13 @@ import {
   AppRouterContext,
   GlobalLayoutRouterContext,
   LayoutRouterContext,
-} from 'next/dist/shared/lib/app-router-context.shared-runtime';
+} from 'next/dist/shared/lib/app-router-context.shared-runtime.js';
 import {
   PathParamsContext,
   PathnameContext,
   SearchParamsContext,
-} from 'next/dist/shared/lib/hooks-client-context.shared-runtime';
-import { PAGE_SEGMENT_KEY } from 'next/dist/shared/lib/segment';
+} from 'next/dist/shared/lib/hooks-client-context.shared-runtime.js';
+import { PAGE_SEGMENT_KEY } from 'next/dist/shared/lib/segment.js';
 
 import type { RouteParams } from './types';
 

--- a/code/frameworks/nextjs/src/routing/decorator.tsx
+++ b/code/frameworks/nextjs/src/routing/decorator.tsx
@@ -2,7 +2,7 @@ import * as React from 'react';
 
 import type { Addon_StoryContext } from 'storybook/internal/types';
 
-import { RedirectBoundary } from 'next/dist/client/components/redirect-boundary';
+import { RedirectBoundary } from 'next/dist/client/components/redirect-boundary.js';
 
 import { AppRouterProvider } from './app-router-provider';
 import { PageRouterProvider } from './page-router-provider';

--- a/code/frameworks/nextjs/src/routing/page-router-provider.tsx
+++ b/code/frameworks/nextjs/src/routing/page-router-provider.tsx
@@ -8,7 +8,7 @@ import React from 'react';
 
 import { getRouter } from '@storybook/nextjs/router.mock';
 
-import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime';
+import { RouterContext } from 'next/dist/shared/lib/router-context.shared-runtime.js';
 
 export const PageRouterProvider: React.FC<PropsWithChildren> = ({ children }) => (
   <RouterContext.Provider value={getRouter()}>{children}</RouterContext.Provider>

